src/dom.js

```diff
@@ -435,16 +375,54 @@ export function childElementByTag(parent, tagName) {
  * @return {!Array<!Element>}
  */
 export function childElementsByTag(parent, tagName) {
+  return scopedQuerySelectorAll(parent, `> ${tagName}`);
+}
+
+
+/**
+ * Finds the first element that matches `selector`, scoped inside `root`.
+ * Note: in IE, this causes a quick mutation of the element's class list.
+ * @param {!Element} root
+ * @param {string} selector
+ * @return {?Element}
+ */
+export function scopedQuerySelector(root, selector) {
```

Comment:
This isn't due to `:scope`, but that there are now two selectors to match against (`.selector1` vs `.selector1 .selector2`).

I'm not requiring `scopedQuerySelector` where we can avoid it (it's fine if you're doing `scope.querySelector('selector')`). The only time it's necessary is when there's a descendant selector included (`scope.querySelector('parent descendant')`), which is already slow.